source/_posts/2019-01-28-building-a-decentralized-reddit-with-embark-part-1.md

```diff
@@ -0,0 +1,192 @@
+
+Let's get right to it!
+
+## Functionality Overview
+
+Alright, let's start off with quickly talking about what exactly it is that we want to build. Obviously, Reddit is a pretty sophisticated platform so we won't be able to rebuild it completely. Instead, we'll be focusing on some key features that will also demonstrate very nicely how Embark can help building such an application.
+
+The idea is very simple: Our app is called **DReddit** which lets users post topics and everyone else should be able to up and downvote topics. A user account is coupled to an Ethereum wallet account. Essentially every wallet account is a valid account for the application and users can authenticate using extensions like Metamask.
+
+Here's what it will look like in action:
+
+[TODO GIF]
+
+Notice that the application isn't particularly pretty. This is because we don't want to focus too much on CSS and Design related topics but rather the functionality itself. Feel free to apply any look and feel you think is appropriate.
+
+Cool, with that out of the way, let's start building!
+
+## Setting up the application
+
+If you've read our guide on [Creating Applications](/docs/create_project.html) or our last tutorial on [Building Smart Contract only apps](/news/2019/01/22/building-smart-contract-only-dapps/), you know that Embark comes with a `new` command to scaffold an application. We're going to do exactly that, but first we need to make sure Embark is installed. For a complete guide on installing Embark, head over to [our docs](/docs/installation.html), otherwise, simply run the following command in your terminal of choice:
+
+<pre><code>$ npm install -g embark@next</code></pre>
+
+Notice that we're installing Embark's development version by using the `@next` tag. This will go away in the near future, however to get the newest features and performance improvements, this is the way to go for now.
+
+Next, we'll create and set up our app using the `new` command:
+
+<pre><code>$ embark new dreddit
+$ cd dreddit
+</code></pre>
+
+Now is a good time to familiarize ourselves with the project structure. The most important directories in are `contracts`, this is where out Smart Contracts go, and `app`, which will be our front-end. Take your time to take a look and check out our [Application Structure](/docs/structure.html) guide for more detailed overview.
+
+Also, to ensure and double-check that everything's working, we can run the application using Embark's `run` command:
+
+<pre><code>$ embark run</code></pre>
+
+If there are any issues in the "Available Services" section of the dashboard, go back to our installation guide and make sure all tools are available on your machine.
+
+## Creating the Smart Contract
+
+Alright, next up we want to create the brain of our application, which is a Smart Contract that enables creating posts and votes. We're going to build it up step by step and afterwards we'll add some tests to ensure our code is actually working.
```

Comment:
Would it be good to link to solidity documentation?

Reply:
added!